go/libkb/merkle_client_test.go

```diff
@@ -0,0 +1,35 @@
+	var r MerkleRootPayloadUnpacked
+	err := json.Unmarshal([]byte(raw), &r)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var h NodeHash
+	h, err = NodeHashFromHex(`fb99c7ba0844733318de3aaea7e19144c4ebc20365b5b90e8c443103c6975bd7`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !hashEq(h, r.Body.Skips[227711]) {
+		t.Fatal("should have had hash equality")
+	}
+}
+
+func TestMerkleRootPayloadUnmarshalWithoutSkips(t *testing.T) {
+	raw := `{"body":{"kbfs":{"private":{"root":"iKNlcGvEICJOVuOROjr8mEPkhETSMBpVZBJ41nBYBuodK/OrXGFaoWjEQINGJ9t9R8Eu4kLPLihrqqB21NDlieEMizSPcogzKeR3Q4Hsmt/uid87xpmvrZc4LhAQOX62sCEf3KTaSWjZG0Gjbm9uxBgfGYN0/evl0mbBib/jafJ/5yJIcDizzhiicHLEQMW85WnignmXsa5mLomnmduzUSksYOxUZU1bvJwn9Isj5Tyw2AL4t4t3nzgF1VEricihIF5N7+Pih/HlO0HAFE6ic27NHi2hdAKidHPOWHespKF2AQ==","version":1},"public":{"root":"hqFoxEBK5PjCumUTQuUrwDPI08l8x+QgiEqq5AbRVQGimEjDozDf6ax5mKh/A3qGv5Sjgxqkni36/rKxE6JBODpXJwJjonByxEBpdeld3+IbKVY7ACqtETV9cXP1UPO+Sxj+zJ2PdmmuelBZIYcW3FgcGiPJm15d/rpnSq2ob4m7u8B1wgWeLpcbonNuzR42oXQBonRzzlh3rJehdgE=","version":1}},"key":{"fingerprint":"03e146cdaf8136680ad566912a32340cec8c9492","key_id":"2A32340CEC8C9492"},"legacy_uid_root":"0c58a74abcb1db656146ebf667b5ea5cc55e208fbb534cd1b1e9454890358e5e","prev":"f82f0d1c236703205690110203e34cd9fc5dfb7a6bef386e81a1c9301b101deea176fae530ca654424cdd72565f1dc196af1eaed203b7dd0a100d672080b1961","root":"976a48126ced7ba136ef1a33e337c9ad22ca233eb19d46a5a1de482653b0ac25e35d3e55f1cb1fc5a29e62df5612581909afe8a26dc44710da5094445f009af0","seqno":796746,"txid":"db7ad0824c5753c28bc849d60e711815","type":"merkle_root","version":1},"ctime":1484239402,"tag":"signature"}`
+	var r MerkleRootPayloadUnpacked
+	err := json.Unmarshal([]byte(raw), &r)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if r.Body.Skips != nil {
+		t.Fatal("expected an empty skip table")
+	}
+}
```

Comment:
Can we add a test for bad skip pointers? Maybe one case where the skip list has a bad hash in it, and one case where it refers to a node that doesn't contain the next pointer?